_posts/2021-07-07-high-level-graphs.md

```diff
@@ -0,0 +1,160 @@
+
+In fact, you might already be benefitting from high level graphs:
+> "Starting with Dask 2021.05.0, Dask DataFrame computations will start sending HighLevelGraph's directly from the client to the scheduler by default. Because of this, users should observe a much smaller delay between when they call .compute() and when the corresponding tasks begin running on workers for large DataFrame computations" https://coiled.io/dask-heartbeat-by-coiled-2021-06-10/
+
+Read on for a snapshot of progress in other areas.
+
+## Blockwise layers progress
+### Summary
+
+The `Blockwise` high level graph layer was introduced in the 2020.12.0 Dask release. Since then, there has been a lot of effort made to use `Blockwise` high level graph layer whereever possible for improved performance, most especially for IO operations. The following is a non-exhaustive list.
+
+### Work to date
+Highlights include (in no particular order):
+
+1. Merged PR by Rick Zamora: [Use Blockwise for DataFrame IO (parquet, csv, and orc) #7415](https://github.com/dask/dask/pull/7415)
+2. Merged PR by Rick Zamora: [Move read_hdf to Blockwise 7625](https://github.com/dask/dask/pull/7625)
+3. Merged PR by John Kirkham: [Move timeseries and daily-stock to Blockwise #7615](https://github.com/dask/dask/pull/7615)
+4. Merged PR by John Kirkham: [Rewrite da.fromfunction w/ da.blockwise #7704](https://github.com/dask/dask/pull/7704)
+
+### Ongoing work
+Lots of other work with `Blockwise` is currently in progress:
+
+1. Ian Rose: [Blockwise array creation redux #7417](https://github.com/dask/dask/pull/7417). This PR creates blockwise implementations for the `from_array` and `from_zarr` functions.
+2. Rick Zamora: [Move DataFrame from_array and from_pandas to Blockwise #7628](https://github.com/dask/dask/pull/7628)
+3. Richard Berry: [Use BlockwiseDep for map_blocks with block_id or block_info #7686](https://github.com/dask/dask/pull/7686)
+
+## A high level graph for `map_overlap`
+### Summary
+
+Investigating a high level graph for Dask's `map_overlap` is a project driven by user needs. People have told us that the time taken just to generate the task graph (before any actual computation takes place) can sometimes be a big user experience problem. So, we're looking in to ways to improve it.
+
+### Work to date
+
+1. Merged PR by Genevieve Buckley: [A HighLevelGraph abstract layer for map_overlap #7595](https://github.com/dask/dask/pull/7595)
+
+This PR defers much of the computation involved in creating the Dask task graph, but does not eliminate it. Further optimization is needed.
+
+### Ongoing work
+
+Followup work includes:
+1. Find number of tasks in overlap layer without materializing the layer #7788 https://github.com/dask/dask/issues/7788
+2. Implement cull method for ArrayOverlapLayer #7789 https://github.com/dask/dask/issues/7789
```

Comment:
Nice, I get this based on our offline discussion, would be nice to explain what culling is, though that would be challenging in a short space...

Reply:
Done